jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/ServeAs.java

```diff
@@ -0,0 +1,21 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security;
+
+import org.eclipse.jetty.server.Request;
+
+public interface ServeAs
+{
+    Request serveAs(Request request, String path);
+}
```

Comment:
Does this really need to be a top level interface?
Could it be `AuthenticationState.SaveAs.Request` instead?

Reply:
Done, but it was very confusing to call it `Request` because it was conflicting with argument and return value. I named it `ServeAs.Path` instead.